wslpy/system.py

```diff
@@ -78,4 +78,67 @@ def shellEnvVar(input):
 
 
 def registry(input, key):
-    raise NotImplementedError
\ No newline at end of file
+
+    #DUMMY CODE BEGINS (i am only using this to illustrate something)
+    cmd = u"reg.exe query \""+input+"\" /v \""+key+u"\" 2>&1"
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    routput, err = p.communicate()
+    #DUMMY CODE ENDS 
+
+    # Given a valid registry path, retrieves the value of an entry in the registry.
+    # Eg: registry("HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment","OS") returns "WINDOWS_NT"
+    #
+    # MAY return string type TODO: Doees this make sense?
+    #
+    # A valid registry path typically includes (this this and this)
+    #
+    # Parameters
+    # ----------
+    # input : str
+    #     string of a shell environment variable key.
+    #   
+    #     Currently expected values: 
+    #     "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment" (for system)
+    #     "HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\User Shell Folders" (for shell)
+    #     (Although any valid entries should work too) TODO: Check if required
+    #
+    # key : str 
+    #     the name of the registry's key in string
+    #
+    # Returns
+    # -------
+    # The corresponding value TODO: Possibly type
+    #
+    # Raises
+    # ------
+    # Returns the error from Reg.exe (Possibly? )
+
+    try:
+        query = __regInfoFetch__(input, key)
+        
+        if query != "value.": #Error after regex #TODO: Check with Patrick on how he wants to handle this
+            return query #Expected one
+        else:
+            raise KeyError("The following error propogated from Registry:" + routput.decode("utf-8").rstrip()) #TODO: 
```

Comment:
(This is where dummy code comes in) 

I was thinking about how to handle invalid entries, and I wasn't sure how I would go about implementing this considering that the result from __regInfoFetch()__ is pre-formatted. Are there plans for this to become a feature?/Should I try to implement this?

Reply:
Notice the command from 83-85:

```
cmd = u"reg.exe query \""+input+"\" /v \""+key+u"\" 2>&1"
p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
routput, err = p.communicate()
```

`p.communicate` will return both the output and error. That might be helpful for you.

Reply:
I had a thought, I'm thinking of amending __regInfoFetch()__ to return either the error or the value like so:

```
def __regInfoFetch__(input, key):
    cmd = u"reg.exe query \""+input+"\" /v \""+key+u"\" 2>&1"
    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
    routput, err = p.communicate()
    output = (routput.decode("utf-8").rstrip().split())[-1]

    if err == None: #No error
        return output
    else:
        return err
```

Actually: would you rather me make the judgement on design decisions like these?

Reply:
Thanks for this, I must've glossed over this for some reason. 

Unfortunately err doesn't seem to bring out errors and I had to implement a workaround (see https://github.com/wslutilities/wslpy/commit/6fc47610d174f222973387da362e454b654ac5f0 ) to handle this. Perhaps this is an issue with upstream Python on handling errors from Powershell via WSL